src/renderer/actions/connections.js

```diff
@@ -61,15 +61,16 @@ export function connect (id, databaseName, reconnecting = false, sshPassphrase)
       dispatch({ type: CONNECTION_REQUEST, server, database, reconnecting, isServerConnection: !databaseName });
 
       if (!serverSession) {
-        if (server.ssh.privateKeyWithPassphrase && typeof sshPassphrase === 'undefined') {
-          dispatch({ type: CONNECTION_REQUIRE_SSH_PASSWORD });
-          return;
+        if (server.ssh) {
```

Comment:
There is no need for this check if you also check it in nested IF statement. Could you please remove upper if and modify line 70 by adding `if ( server.ssh && ...)`. Like Gareth did in his PR before he removed it :)